WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsTimezoneDialog.java

```diff
@@ -0,0 +1,338 @@
+        dismiss();
+    }
+
+    @Override
+    public void onDismiss(DialogInterface dialog) {
+        Fragment target = getTargetFragment();
+        if (mConfirmed && target != null && !TextUtils.isEmpty(mSelectedTimezone)) {
+            Intent intent = new Intent().putExtra(KEY_TIMEZONE, mSelectedTimezone);
+            target.onActivityResult(getTargetRequestCode(), Activity.RESULT_OK, intent);
+        }
+
+        super.onDismiss(dialog);
+    }
+
+    private void showEmptyView(boolean show) {
+        if (isAdded()) {
+            mEmptyView.setVisibility(show ? View.VISIBLE : View.GONE);
+        }
+    }
+
+    private void showProgressView(boolean show) {
+        if (isAdded()) {
+            mProgressView.setVisibility(show ? View.VISIBLE : View.GONE);
+        }
+    }
+
+    private void hideSearchKeyboard() {
+        if (isAdded()) {
+            ActivityUtils.hideKeyboardForced(mSearchView);
+        }
+    }
+
+    private class TimezoneViewHolder {
+        private final TextView txtLabel;
+
+        TimezoneViewHolder(View view) {
+            txtLabel = view.findViewById(android.R.id.text1);
+        }
+    }
+
+    private class TimezoneAdapter extends BaseAdapter implements Filterable {
+        private final List<Timezone> mAllTimezones = new ArrayList<>();
+        private final List<Timezone> mFilteredTimezones = new ArrayList<>();
+
+        private TimezoneAdapter(@NonNull List<Timezone> timezones) {
+            mAllTimezones.addAll(timezones);
+            mFilteredTimezones.addAll(timezones);
+        }
+
+        @Override
+        public int getCount() {
+            return mFilteredTimezones.size();
+        }
+
+        @Override
+        public Object getItem(int position) {
+            return mFilteredTimezones.get(position);
+        }
+
+        @Override
+        public long getItemId(int position) {
+            return position;
+        }
+
+        public int indexOfValue(String timezoneValue) {
+            if (!TextUtils.isEmpty(timezoneValue)) {
+                for (int i = 0; i < mFilteredTimezones.size(); i++) {
+                    if (timezoneValue.equals(mFilteredTimezones.get(i).value)) {
+                        return i;
+                    }
+                }
+            }
+            return -1;
+        }
+
+        @Override
+        public View getView(int position, View convertView, ViewGroup parent) {
+            TimezoneViewHolder holder;
+            if (convertView == null || convertView.getTag() == null) {
+                convertView = mInflater.inflate(android.R.layout.simple_list_item_1, parent, false);
+                holder = new TimezoneViewHolder(convertView);
+                convertView.setTag(holder);
+            } else {
+                holder = (TimezoneViewHolder) convertView.getTag();
+            }
+
+            boolean isSelected = mSelectedTimezone != null && mSelectedTimezone.equals(mFilteredTimezones.get(position).value);
```

Comment:
Nitpick: Do you mind if we separate this into 2 lines to make it less than 120 chars?